### ISSUE TITLE
Let expressions cannot be used in a conditional expression without parenthesis

### ISSUE THREAD

**Issue by pubudu91** [ballerina-platform/ballerina-lang#34108]
Consider the following:
```ballerina
int b = true ? let int z = 0 in z : let int y = 1 in y; // syntax error
int c = true ? (let int z = 0 in z) : (let int y = 1 in y); // fine
```

**Comment by lochana-chathura** [comment]
Related issue #33215

**Comment by lochana-chathura** [comment]
Fixed in #34313.
